src/optimizer/filter_push_down_optimizer.cpp

```diff
@@ -139,6 +140,26 @@ static ColumnPredicateSet getPropertyPredicateSet(const Expression& property,
     return propertyPredicateSet;
 }
 
+// TODO need determine whether the expression is a constant based on its output type
+static bool isLiteralExpression(const std::shared_ptr<Expression> expression) {
+    switch (expression->expressionType) {
+    case ExpressionType::LITERAL:
+    case ExpressionType::PARAMETER: {
+        return true;
+    }
+    case ExpressionType::FUNCTION: {
```

Comment:
Can u leave a TODO for me here.
```
TODO(Xiyang): fold parameter expression in binder.
```

`CAST($1)` should ideally be folded once we obtain `$1` from user input. I'm fairly certain `CAST("1", INT64)` has already been folded to `1` in current master. We just need to propagate this to parameter and then we don't need to do this check recursively.